src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -53,10 +54,23 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
         _attrs(attrs),
         _pd(pd) {
             _enable_profiling = config.get_property(ov::enable_profiling);
+            _scratchpad_md = _pd.scratchpad_desc();
+
             GPU_DEBUG_GET_INSTANCE(debug_config);
             GPU_DEBUG_IF(!debug_config->dump_profiling_data.empty()) {
                 _enable_profiling = true;
             }
+
+            GPU_DEBUG_IF(debug_config->verbose >= 5) {
+                if (_scratchpad_md.get_size() > 0) {
+                    static long long total = 0;
+                    int64_t size = _scratchpad_md.get_size() / 1048576;
+                    total += size;
```

Comment:
That's thread unsafe code

Reply:
It is changed to atomic variable. Anyway, I think thread-safe is not important because it is for debugging feature.